### PR TITLE
Update to chrono-node v2.3.2

### PR DESCRIPTION
**Pull request by jayaddison-collabora**
#### Summary
Updates the `chrono-node` dependency to [v2.3.2](https://github.com/wanasit/chrono/releases/tag/v2.3.2).

### CODE REVIEW

**Comment by jayaddison-collabora** [comment]
After discovering https://github.com/wanasit/chrono/issues/408 during local deployment, I think that should be resolved before this issue can be merged.

**Comment by emilyacook** [comment]
Thanks for participating in Hacktoberfest! You can claim your sticker set here: https://get.printfection.com/hacktober21/4144583267 @jayaddison-collabora

**Comment by jayaddison-collabora** [comment]
There's a possible issue with one of the test cases in this plugin that I'd like to investigate before merge.

Currently tests pass if `TZ=UTC` is set in the environment, but they do not pass for me locally without that.

The test that passes does include a post timestamp and an origin timezone, so I'm wondering if this is a sign of a potentially unstable timezone conversion (related to daylights savings changes, in particular).

```
  ● timezoneParsing

    expect(received).toEqual(expected) // deep equality

    Expected: "The game is `at 12pm UTC` *(1:00 PM BST)*"
    Received: "The game is `at 12pm UTC` *(12:00 PM BST)*"

      62 | 
      63 |     testCases.forEach((tc) => {
    > 64 |         expect(convertTimesToLocal(tc.test, 1629738610000, 'Europe/London', 'en')).toEqual(tc.expected);
         |                                                                                    ^
      65 |     });
      66 | });
      67 | 

      at toEqual (src/test.js:64:84)
          at Array.forEach (<anonymous>)
      at Object.forEach (src/test.js:63:15)
```

File: webapp/src/time.js [hunk 1]
```diff
@@ -31,12 +47,12 @@ export function convertTimesToLocal(message, messageCreationTime, localTimezone,
 
         let formattedDisplayDate;
 
-        const currentUserStartDate = moment(parsedTime.start.date()).tz(localTimezone).locale(locale);
+        const currentUserStartDate = chronoToMoment(parsedTime.start).tz(localTimezone).locale(locale);
```
Comment (jayaddison-collabora) [inline webapp/src/time.js:50]: @wanasit There's some strange behaviour I've noticed with `chrono-node` 2.x around the `date()` function on result objects.

Currently I'm in a timezone where daylight savings finished on Oct 31st this year.

[One of the test cases](https://github.com/jayaddison-collabora/mattermost-plugin-walltime/blob/1d4d56eacdbba2652d223b1effbeea1c3709a8bb/webapp/src/test.js#L23) in this repository references a date where daylight savings _was_ active (Aug 23rd).

Using the `date()` function as-is to export a `Date` seems to result in an object that assumes local time (2021-08-23 12:00) -- which then gets adjusted to 2021-08-23 11:00 UTC (to account for the daylight savings which JavaScript believes are in effect).

This _might_ be something to do with the [Reference API change re: system time handling](https://github.com/wanasit/chrono/commit/ea0d1717843cd425361d86056d76e0b57e3ff6df#diff-1c0edb2e88e4f027868f0b0389e98b0ee10cf56d0949c931f9b5f2c7831bc191L126-R145) -- I'm not 100% sure yet.

As a workaround here I'm retrieving enough of the individual fields from the result object (using `result.get`) to construct an object that stores the timezone offset detected by `chrono-node`.

Comment (wanasit) [reply webapp/src/time.js:50]: Yes. There was the change that the timezone adjustment when transforming the result into date object at the time. 

Could you try updating to `v2.3.4`? I tried to make the calculation similar [in this change](https://github.com/wanasit/chrono/commit/543c3bccb2f9b0216e3f1d6bf42d16491558e9cc). However, note that:
* This is not a complete fix. On the edge-cases (e.g. near mid-night on October 31/November 1), the result could still be wrong.
* There is no easy way to test the `date()` function in Chrono. The behavior/output of that function is depending on your machine locale setting (that why there is no unit tests).
* Because of the Javascript edge-case, this also not works before 1970 (see the commit).

You may want to consider keeping the workaround you mentioned.

Comment (jayaddison-collabora) [reply webapp/src/time.js:44]: Thank you again @wanasit - the upgrade to v2.3.4 works well and solves the `now` text parsing issue :+1: 

The code would be cleaner without the `chronoToMoment` workaround, and thanks to the change you mention, the workaround could be removed when using v2.3.4, and tests would still pass (but I haven't committed that as a change, yet).

I'll spend a bit more time considering the edge cases and the tradeoffs related to keeping the workaround.

Comment (jayaddison-collabora) [reply webapp/src/time.js]: My thoughts on the tradeoffs: I'm not smart enough to determine whether the workaround helps to reduce many/any other edge case instances, and I think that we can rely on our user community to report issues when they arise.

That's fairly neutral, so I think that another factor to mention is that less code means less for future maintainers to understand and debug through if & when they are investigating issues.

So based on those two considerations, I've removed the `chronoToMoment` workaround.  As a side-effect, hopefully it makes code review for this pull request easier, too.

**Comment by jayaddison-collabora** [comment]
@mkraft There have been a few updates and a bit of cleanup here (mostly introduction, and then removal, of a couple of workarounds for issues discovered during further testing).  Would it be possible to re-review the pull request?

**Review by mkraft — APPROVED** [review]

**Comment by jayaddison-collabora** [comment]
@mkraft Let me know if there's anything else I could/should do here before the possibility of merge.  I think there will be some small merge conflicts to resolve between this and #41, or vice-versa depending on merge order (and I'm happy to be available to sort those out).

**Comment by jayaddison-collabora** [comment]
Thank you for the reviews and merge @mkraft!  This has been merged back from mainline into #41, and conflicts resolved.
